snuba/datasets/entities/metrics.py

```diff
@@ -104,6 +105,7 @@ def get_extensions(self) -> Mapping[str, QueryExtension]:
     def get_query_processors(self) -> Sequence[QueryProcessor]:
         return [
             BasicFunctionsProcessor(),
+            GranularityProcessor(),
```

Comment:
So, if my understanding is correct, only a limited amount of granularities will be allowed in the query.
The ones we have in the DB and all their multiples. 
This processor picks the granularity and sets a condition as they are all in the same storage, then The TimeSeriesProcessor applies the proper time transformation.
You will never use the bare `timestamp` column. 

Is this correct ?